### PR TITLE
Replace black with 'ruff format'

### PR DESCRIPTION
**Pull request by anz-ableton**

### CODE REVIEW

File: requirements-dev.txt [hunk 1]
```diff
@@ -1,34 +1,21 @@
 #
-# This file is autogenerated by pip-compile with Python 3.10
+# This file is autogenerated by pip-compile with Python 3.12
```
Comment (nre-ableton) [inline requirements-dev.txt:2]: Please move this change to a separate commit.

**Review by nre-ableton — CHANGES_REQUESTED** [review]

File: requirements-dev.txt (hunk 1 above)
Comment (anz-ableton) [reply requirements-dev.txt:2]: The requirements-dev.txt update got moved into 9c44189e60e46abac8b335485ea14c290a9d9b82.

**Review by nre-ableton — APPROVED** [review]
